core/src/main/java/io/grpc/ChannelImpl.java

```diff
@@ -195,12 +204,13 @@ public ChannelImpl shutdown() {
    * Initiates a forceful shutdown in which preexisting and new calls are cancelled. Although
    * forceful, the shutdown process is still not instantaneous; {@link #isTerminated()} will likely
    * return {@code false} immediately after this method returns.
-   *
-   * <p>NOT YET IMPLEMENTED. This method currently behaves identically to shutdown().
    */
-  // TODO(ejona86): cancel preexisting calls.
   public ChannelImpl shutdownNow() {
     shutdown();
+
+    for (ClientCallImpl<?, ?> call : activeCalls) {
+      call.getStream().cancel(Status.INTERNAL.withDescription("Shutting down"));
```

Comment:
I think it may be time to just make ClientCalls thread safe.  In the past there was no need, but now there is.

If you feel there is too high of a cost to synchronizing Call, it will push the synchronization on the the application (the one that invoked shutdownNow), and just say that shutdownNow is not thread safe.  In this case, we can just say that shutdown is the preferred method.